packages/itwinui-react/src/core/Tree/Tree.tsx

```diff
@@ -380,11 +403,29 @@ const VirtualizedTree = React.forwardRef(
     }: VirtualizedTreeProps<T>,
     ref: React.ForwardedRef<HTMLUListElement>,
   ) => {
-    const { outerProps, innerProps, visibleChildren } = useVirtualization({
-      itemsLength: flatNodesList.length,
-      itemRenderer: itemRenderer,
+    const parentRef = React.useRef<HTMLDivElement | null>(null);
+
+    const virtualizer = useVirtualScroll(
```

Comment:
Performance seems to be the same as the old implementation across Edge, Chrome, and Firefox. CPU throttling to 20x does slow the virtualized `Tree` extremely, but this also seems to be the case with the old implementation. Video: 

https://github.com/iTwin/iTwinUI/assets/141063311/b849f612-f97d-4c23-a9a0-b700b1b82b32

I have only been able to test with the Windows OS, and as a result, have not been able to test performance with Safari. A high `overscan` value would certainly have a negative effect on performance, but the value `useVirtualScroll` uses, 10, seems to not have much impact. `estimateSize` likely doesn't have much impact on performance itself, since elements are always measured when they are first rendered, regardless of the estimate itself. I did find that the measurements aren't updated the size of the items, however. I am investigating this issue currently. Screenshot with larger estimate: 
![image](https://github.com/iTwin/iTwinUI/assets/141063311/fc5b53bd-afd3-44c4-afe5-e38990b80d82)

 Unrelated: I did discover an issue with Firefox while testing where scrolling too quickly would cause the virtualized tree to flicker. This happens both on the old and new implementation. I will investigate this more and see if I can find a fix. Video: 

https://github.com/iTwin/iTwinUI/assets/141063311/b1a47362-8991-412f-a938-259cf447122d

Reply:
An update on the issues talked about here, the dynamic sizing seems to have been fixed by a different change. I think this [PR](https://github.com/iTwin/iTwinUI/pull/2111) is responsible for the fix, since tanstack virtual's dynamic sizing is reliant on the `virtualizer.measureElement` ref being passed into the `TreeNode`s correctly. Did some performance testing, and found that the estimate being further from the actual size of the items causes worse performance in tanstack virtual. Video of performance with estimate size returning `50px`: 

https://github.com/iTwin/iTwinUI/assets/141063311/06ce6ba9-d4e2-4425-807c-00ebe1194ba3

Video with `39px` estimate:

https://github.com/iTwin/iTwinUI/assets/141063311/c62f4940-4242-4507-8604-0dcf85b3a5f4

Both of these videos are with 20x CPU throttling. This reinforces that we should keep the estimate at around `39px`, since that will be the most common usage of the virtualize `Tree`. Additionally the Firefox issue is partially resolved with only the text flickering on fast scroll, instead of the entire `Tree`. I think changing the `<ul>` to `<div>` fixed this. The flickering text is obviously not ideal, but it only happens when scrolling very quickly with the scrollbar. Video: 

https://github.com/iTwin/iTwinUI/assets/141063311/9b0c583a-128f-4305-857e-b5c70faadef7

Reply:
Updated `estimateSize` to return the measured height of the first `TreeNode` [here](https://github.com/iTwin/iTwinUI/pull/2091/commits/6200d585c8b3c712c60a05e7d045190193dc5a4b). I did test having the default return `50px` instead of `39px` to see both if this worked and if it improved performance in the case of the item being a different size from the estimate, and it seemed to do both.

Reply:
I think figuring out which nodes have sublabels is going to be a problem, since we allow users to define the type of data that they are passing to the tree, which may not even have a sublabel field. I'll return it to using 39px for now. No, I am not suggesting we change the fall back estimate to 50px, I was just using that as a test for a fallback estimate that was very different from the actual size of the `TreeNode`s. By "it", I meant using the measured height of the first item, not using a size estimate of 50px.

Reply:
Removed code measuring first item's height and returned to using `39px` as estimate in this [commit](https://github.com/iTwin/iTwinUI/pull/2091/commits/788973864919acc8134faa264c6fe40c2368414b).